File: LEETCODE/arrays/Medium/S4_Leetcode_53MaxsubArrayKadan.java
package LEETCODE.arrays.Medium;

import LEETCODE.arrays.ArrayInput;

public class S4_Leetcode_53MaxsubArrayKadan {
    public static int  kadansAlgo(int[] nums){
        int n = nums.length;
        int sum = 0;
        int maxsum = Integer.MIN_VALUE;
        for(int i = 0; i < n; i++){
            sum = sum + nums[i];
            maxsum = Math.max(maxsum, sum);
            if(sum < 0){
                sum = 0;
            }
        }
        return maxsum;
    }

    public static void main(String[] args) {
        int [] nums = ArrayInput.takeArrayInput();
        kadansAlgo(nums);
    }
}

File: LEETCODE/arrays/Medium/S4_Leetcode_53MaxsubArrayKadan.cpp
#include<bits/stdc++.h>
using namespace std;

int kadans(vector<int> nums){
    int n = nums.size();
    int cs = 0;
    int maxsum = INT_MIN;
    for(int i = 0; i < n; i++){
        cs += nums[i];
        maxsum = max(cs,maxsum);
        if(cs < 0){
            cs = 0;
        }
    }
    return maxsum;
}

int main(){
    int n;
    cin >> n;
    vector<int> v(n);
    for(auto i : v){
        cin >> i;
    }
}